src/shapes/image.class.ts

```diff
@@ -1,869 +1,797 @@
-          filters,
-          this._originalElement,
-          sourceWidth,
-          sourceHeight,
-          this._element,
-          this.cacheKey
-        );
-        if (
-          this._originalElement.width !== this._element.width ||
-          this._originalElement.height !== this._element.height
-        ) {
-          this._filterScalingX =
-            this._element.width / this._originalElement.width;
-          this._filterScalingY =
-            this._element.height / this._originalElement.height;
+      }
+      if (pAR.meetOrSlice === 'slice') {
+        scaleX = scaleY = findScaleToCover(this._element, parsedAttributes);
+        offset = rWidth - pWidth / scaleX;
+        if (pAR.alignX === 'Mid') {
+          cropX = offset / 2;
         }
-        return this;
-      },
-
-      /**
-       * @private
-       * @param {CanvasRenderingContext2D} ctx Context to render on
-       */
-      _render: function (ctx) {
-        ctx.imageSmoothingEnabled = this.imageSmoothing;
-        if (
-          this.isMoving !== true &&
-          this.resizeFilter &&
-          this._needsResize()
-        ) {
-          this.applyResizeFilters();
+        if (pAR.alignX === 'Max') {
+          cropX = offset;
         }
-        this._stroke(ctx);
-        this._renderPaintInOrder(ctx);
-      },
-
-      /**
-       * Paint the cached copy of the object on the target context.
-       * it will set the imageSmoothing for the draw operation
-       * @param {CanvasRenderingContext2D} ctx Context to render on
-       */
-      drawCacheOnCanvas: function (ctx) {
-        ctx.imageSmoothingEnabled = this.imageSmoothing;
-        FabricObject.prototype.drawCacheOnCanvas.call(this, ctx);
-      },
-
-      /**
-       * Decide if the object should cache or not. Create its own cache level
-       * needsItsOwnCache should be used when the object drawing method requires
-       * a cache step. None of the fabric classes requires it.
-       * Generally you do not cache objects in groups because the group outside is cached.
-       * This is the special image version where we would like to avoid caching where possible.
-       * Essentially images do not benefit from caching. They may require caching, and in that
-       * case we do it. Also caching an image usually ends in a loss of details.
-       * A full performance audit should be done.
-       * @return {Boolean}
-       */
-      shouldCache: function () {
-        return this.needsItsOwnCache();
-      },
-
-      _renderFill: function (ctx) {
-        var elementToDraw = this._element;
-        if (!elementToDraw) {
-          return;
+        offset = rHeight - pHeight / scaleY;
+        if (pAR.alignY === 'Mid') {
+          cropY = offset / 2;
         }
-        var scaleX = this._filterScalingX,
-          scaleY = this._filterScalingY,
-          w = this.width,
-          h = this.height,
-          min = Math.min,
-          max = Math.max,
```

Comment:
this was purely done to lend an hand to the minifier.
And i occasionally do, but i also think probably terser is smart enough to do it itself?

Reply:
I understand 
Too messy IMO